snapcraft/cli/lifecycle.py

```diff
@@ -117,6 +117,10 @@ def _execute(  # noqa: C901
         if pack_project:
             _pack(project.prime_dir, output=output)
     else:
+        # load project and validate schema before invoking a provider
```

Comment:
in the comment you mention having rolled back this logic, but I still see it here?

Reply:
Commits split at the wrong point. Re-splitting.